integrationtests/src/test/java/org/neo4j/causalclustering/scenarios/ListQueriesProcedureInClusterIT.java

```diff
@@ -0,0 +1,172 @@
+            try {
+
+                //Given
+                ReadReplicaGraphDatabase replicaDb = cluster.findAnyReadReplica().database();
+                Node node = leaderDb.createNode();
+                tx.success();
+
+                threads.execute(parameter ->
+                {
+                    cluster.coreTx((leaderDb1, secondTransaction) -> {
+
+                        //lock node
+                        lock.accept(secondTransaction, node);
+                        resourceLocked.countDown();
+                        try {
+                            listQueriesLatch.await();
+                        } catch (InterruptedException e) {
+                            e.printStackTrace();
+                            fail("failure in locking node");
+                        }
+                    });
+                    return null;
+                }, null);
+
+                resourceLocked.await();
+
+                //When
+                //Execute query on Leader
+                threads.execute(parameter ->
+                {
+                    leaderDb.execute(CORE_QUERY).close();
+                    return null;
+                }, null);
+
+                //Execute query on Replica
+                threads.execute(parameter ->
+                {
+                    replicaDb.execute(REPLICA_QUERY).close();
+                    return null;
+                }, null);
+
+                //Then
+                try  {
+                    Map<String, Object> coreQueryListing = getQueryListing(CORE_QUERY, leaderDb).get();
+                    Map<String, Object> replicaQueryListing = getQueryListing(REPLICA_QUERY, replicaDb).get();
+
+                    assertNotNull(coreQueryListing);
+                    assertNotNull(replicaQueryListing);
+
+                    assertThat(coreQueryListing.get("activeLockCount"), is(1l));
+                    assertThat(replicaQueryListing.get("activeLockCount"), is(1l));
+
+                    assertFalse(getQueryListing(REPLICA_QUERY, leaderDb).isPresent());
+                    assertFalse(getQueryListing(CORE_QUERY, replicaDb).isPresent());
+
+                    listQueriesLatch.countDown();
+
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    fail("Couldn't countdown listqueries latch");
+                }
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+                fail("Should create node and list queries");
+            }
+        });
+    }
+
+    private Optional<Map<String, Object>> getQueryListing(String query, GraphDatabaseFacade db) {
+        try (Result rows = db.execute("CALL dbms.listQueries()")) {
+            while (rows.hasNext()) {
+                Map<String, Object> row = rows.next();
+                if (query.equals(row.get("query"))) {
+                    return Optional.of(row);
+                }
+            }
+        }
+        return Optional.empty();
```

Comment:
I wouldn't have written it like this, I would have simply let the method return `Map<String,Object>` instead of an `Optional` thereof, and then invoke `fail(...)` at the end if no matching query could be found. That way it would be easy to provide a good failure message (`NoSuchElementException("No value present")` - which is what `Optional.empty().get()` produces - is not a good failure message).